docs/ERROR_HANDLING.md

```diff
@@ -84,3 +84,48 @@ The following table shows the new handling logic.
 | `ibc_packet_receive`  | ⏮️ state reverted<br>✅ tx succeeds with error ack | ⏮️ state reverted<br>❌ tx fails              |
 | `ibc_packet_ack`      | ⏮️ state reverted<br>❌ tx fails                   | ⏮️ state reverted<br>❌ tx fails              |
 | `ibc_packet_timeout`  | ⏮️ state reverted<br>❌ tx fails                   | ⏮️ state reverted<br>❌ tx fails              |
+
+## Error acknowledgement formatting
+
+In case of a contract error in `ibc_packet_receive`, wasmd creates an error
+acknowledgement. The format used is a JSON object with a single top level
+`error` string such as `{"error":"some error text"}`. This format is the JSON
+serialization of the ibc-go
+[Acknowledgement](https://github.com/cosmos/ibc-go/blob/v7.0.0/proto/ibc/core/channel/v1/channel.proto#L156-L162)
+type and compatible with ICS-20.
+
+If you are using the acknowledgement types shipped with cosmwasm-std
+([#1512](https://github.com/CosmWasm/cosmwasm/issues/1512)), your protocol's
+acknowledgement is compatible with that.
+
+If you are using a customized acknowledgement type, you need to convert contract
+errors to error acks yourself in `ibc_packet_receive`. The `Never` type provides
```

Comment:
This seem to be scenario 2 where the state is committed.
In order to achieve scenario 3, the contract would have to use submessages as in Ethan's [demo](https://github.com/confio/cw-ibc-demo/pull/20).

Reply:
What this does is: never create a contract error. Then `IBCReceiveResult.Err` is never set. The two cases left in wasmd are:
1. Write the acknowledgement from the contract to state
2. A VM error fails the trainsaction (wasmd 0.32+)